src/components/ElementSearchDialog/element-search-dialog.tsx

```diff
@@ -0,0 +1,170 @@
+
+    const displayedValue = useMemo(() => {
+        return searchTermDisabled || searchTermDisableReason
+            ? searchTermDisableReason
+            : searchTerm;
+    }, [searchTerm, searchTermDisabled, searchTermDisableReason]);
+
+    const handleClose = useCallback(() => {
+        onSearchTermChange('');
+        onClose();
+    }, [onSearchTermChange, onClose]);
+
+    return (
+        <Dialog
+            open={open}
+            onClose={handleClose}
+            disableRestoreFocus={true}
+            aria-labelledby="dialog-title-search"
+            fullWidth={true}
+        >
+            <DialogContent>
+                <Autocomplete
+                    id="element-search"
+                    forcePopupIcon={false}
+                    fullWidth
+                    freeSolo
+                    onInputChange={(
+                        _event: React.SyntheticEvent,
+                        value: string,
+                        reason: AutocompleteInputChangeReason
+                    ) => {
+                        // if reason is equal to "reset", it means it programmatically changed (by selecting a result)
+                        // we don't want to change "searchTerm" when clicking a result
+                        if (!searchTermDisabled && reason !== 'reset') {
+                            onSearchTermChange(value);
+                        }
+                    }}
+                    onChange={(
+                        _event: React.SyntheticEvent,
+                        newValue: string | EquipmentInfos | null,
+                        reason?: AutocompleteChangeReason
+                    ) => {
+                        // when calling this method with reason == "selectOption", newValue can't be null or of type "string", since an option has been clicked on
+                        if (
+                            newValue != null &&
+                            typeof newValue !== 'string' &&
+                            reason === 'selectOption'
+                        ) {
+                            onSelectionChange(newValue);
+                        }
+                    }}
+                    getOptionLabel={(option: EquipmentInfos | string) =>
+                        typeof option === 'string' ? option : option.label
+                    }
+                    isOptionEqualToValue={(
+                        option: EquipmentInfos | string,
+                        value: EquipmentInfos | string
+                    ) => {
+                        if (
+                            typeof option === 'string' ||
+                            typeof value === 'string'
+                        ) {
+                            return option === value;
+                        } else {
+                            return option.id === value.id;
+                        }
+                    }}
+                    options={isLoading ? [] : elementsFound}
+                    loading={isLoading}
```

Comment:
If you add `loading` please add `loadingText` as well and some keys for translation purpose

Reply:
I'm not sure I understand, you don't want 
` noOptionsText={intl.formatMessage({
       id: 'element_search/noResult',
  })}`
to be in commons-ui anymore ?

Reply:
I'm talking about `loadingText` (which is by default "Loading..." ) not `noOptionsText`.
It will not be translated in french by MUI, I think we must manage this prop with `loading` prop